Nerfs backpacks

quardbreak opened the pull request:
This PR do, or would do if this idea will supported.
🆑 eckff
tweak: You can't interact with backpack while you wear it on your back. You need take it off if you want put something inside or take out from it. This don't affect to satchels and pocketbooks.
/🆑

CakeQ commented:
Finally
edit: Can you still put laser carbines in backpacks/satchels? If so please fix

PsiOmegaDelta commented on code/game/objects/items/weapons/storage/backpack.dm:
```diff
@@ -22,20 +22,43 @@
+
+/obj/item/weapon/storage/backpack/proc/worn_check(var/mob/L)
+	if(!worn_access && L.back == src)
+		if (istype(L))
```
The line above would most likely have runtimed already if L wasn't the given type.

PsiOmegaDelta commented on code/game/objects/items/weapons/storage/backpack.dm:
```diff
@@ -22,20 +22,43 @@
+	if(!worn_access && L.back == src)
+		if (istype(L))
+			to_chat(L, "<span class='warning'>You need take off \the [src] before you can use it!</span>")
+		if (use_sound)
```
Why does a check have side-effects like playing sounds?

quardbreak replied:
To reduce copy-paste in every function, when we can trigger sound just in one proc?

afterthought2 replied:
Perhaps Psi was unclear: you need to do this elsewhere, for example in `open` several lines above.

PsiOmegaDelta commented on code/game/objects/items/weapons/storage/backpack.dm:
```diff
@@ -227,6 +246,8 @@
 	name = "satchel"
 	desc = "A trendy looking satchel."
 	icon_state = "satchel-norm"
+	max_storage_space = DEFAULT_LARGEBOX_STORAGE
```
Satchels are mobile crates now?

Aticius commented:
Since I can't seem to find who to ping on #feedback,

I personally think an overall storage nerf isn't very welcome, I'd rather see backpacks be buffed, along with dufflebags as the bulk storage options, rather than making satchels and the like smaller.

This game needs less inventory kerfluffling, not more.

Roland410 commented:
Solid no. This is not CM and we don't need bad memes from there. Most jobs already need to use a backpack already to perform efficiently enough along with belts and webbings. If anything, storage needs to be reworked to be less stupid, not make it more annoying.

Sbotkin commented:
Personally I would remove backpacks from the game because wearing a backpack all the time unless you are an engineer or maybe a cargo tech is beyond retarded and bad for your back. This is a move to the right direction :+1:

Roland410 commented:
Sometimes realism has to be put below gameplay. When you can't properly treat someone so you can get them to medbay, or can't store more than two tapes in your coat it's really annoying. If the whole storage system wouldn't be this bad, I wouldn't mind, but unless someone reworks it which probably won't happen anytime soon, backpacks are necessary.

Alex6511 commented:
If this is going to be implemented we need better ways to handle inventory interaction in general first. The system is way too clunky as it stands for this.

Additionally this significantly complicates medbay gameplay.

afterthought2 commented on code/game/objects/items/weapons/storage/backpack.dm:
```diff
@@ -22,20 +22,44 @@
+/obj/item/weapon/storage/backpack/proc/worn_check(var/mob/L)
+	if(!istype(L))
+		return FALSE
+	if(!worn_access && L.back == src)
```
The correct way to check this is `L.get_equipped_item(slot_back)`.

afterthought2 commented on code/game/objects/items/weapons/storage/backpack.dm:
```diff
@@ -22,20 +22,44 @@
+/obj/item/weapon/storage/backpack/open(mob/user)
+	if (!worn_check(user))
+		return
+	if (use_sound)
```
Actually, I just looked it up; this is already done in parent's `open` Should probably just move it back to `attackby` after then new check, or drop it altogether.

babydoll commented:
Do not merge, should get some input from all devs

afterthought2 commented:
Will need a changelog, and possibly wiki updates, if merged.

comma commented:
I think making backpacks bigger instead of making satchels smaller would be better. Wouldn't break any gameplay stuff that depends on current storage volumes (I think engineers aren't even worst offenders, corpsmen are with bags/rollers), and such changes go down easier with smol buffs.
It would accomplish same effect of making backpacks used when you really need bigger storage, but with less pain inflicted.

CakeQ commented:
@comma That doesn't change how current satchels can hold a ridiculous amount at the moment, though

Roland410 commented:
Satchels can't even hold that much stuff really, but if they don't have the pocketbook treatment you could apply that to it.

Commit:
Nerfs backpacks


Tweaks after recived feedback




Fixes PR code


Buff satchels


Another sick change to make everything okay

Anticept commented:
Rather than having to go through taking the backpack off, I'd rather just see a delay in taking something out of it. I think afterthought pointed out a possibility in discord, the fuller the backpack, the longer it takes to ruffle through. That might be alright too.

It just would be really annoying having to toss it on the ground.

Satchels could be something where it's small, and easy to go through so you don't get much of a delay.

Duffels would make 100% sense to have to drop em on the ground though.

quardbreak commented:
@Anticept You don't need toss it on the ground, you can just take backpack in hands and click on it with with another hand.

Anticept commented:
> 
> 
> @Anticept You don't need toss it on the ground, you can just take backpack in hands and click on it with with another hand.

It's still the whole thing of taking it off. It's amazingly *annoying* to do that.

afterthought2 commented:
As discussed on discord, while I'm in favor of some changes in this direction, there are better (or at least less drastic) ways of doing this that mitigate the impact on departments relying heavily on transport and annoy players less. 

I'm not particularly swayed by the downvotes (that's expected for a nerf like this), but some pretty good feedback and suggestions have been made, and I'd like to see some of that acted upon here.

HeyBanditoz commented:
>I'm not particularly swayed by the downvotes...

![screenshot_20181209-233918](https://user-images.githubusercontent.com/7574664/49714826-af425580-fc0b-11e8-8724-9a212336be45.png)
Perhaps another dev would be, then.

babydoll commented:
@eckff 

This won't be merged in its current state and it's not fair to ask you to do something you aren't interested in doing, so let us know if you're interested in taking this further based on recommendations that occurred in #feedback.

quardbreak commented:
Okay I just let this thing die for some time, maybe next time.
